.circleci/config.yml

```
@@ -25,10 +26,10 @@ commands:
       - run:
           name: Setup python
           command: |
-            sudo apt-get update && sudo apt-get install -y python3-dev python3-pip python3-venv
-            python3 -mvenv venv
-            . venv/bin/activate
+            sudo apt-get update && sudo apt-get install -y python3-dev python3-pip
             pip3 install flake8 hawkauthlib konfig pyramid pyramid_hawkauth requests simplejson unittest2 WebTest WSGIProxy2
+            # NOTE: Python3.12 requires `--break-system-packages`.
+            # This command is run on the circleci/mysql:5.7-ram image, which is running python 3.10
```

Again we don't run anything of ours in the mysql image, these run in the job's primary rust image. You can see that in [this branch's](https://github.com/mozilla-services/syncstorage-rs/compare/bug/SYNC-4298_pip...bug/SYNC-4298_pip-pjenvey) ci run [here](https://app.circleci.com/pipelines/github/mozilla-services/syncstorage-rs/5162/workflows/ce939a49-c623-4407-a81e-60b1d08fb706/jobs/20052/parallel-runs/0/steps/0-103) and [here](https://app.circleci.com/pipelines/github/mozilla-services/syncstorage-rs/5162/workflows/ce939a49-c623-4407-a81e-60b1d08fb706/jobs/20052/parallel-runs/0/steps/0-110)
```suggestion
            # This command is run on the cimg/rust image, which is running python 3.10
```

---

Yeah, considering there's a docker image that's created that runs a docker compose that brings up two other docker images, I need a road map to figure out all the dependencies.

---

There's no explicit version of Python, but one is usually built into whatever base image that the docker image is built from. In our case, `cimg/rust` is built off of an image that uses `Ubuntu 22.04`, which pre-installs Python 3.10 (with Python 3.11 available for install). 22.04 was [the prior LTS](https://wiki.ubuntu.com/Releases) for Ubuntu, with 24.04 being the current. I suspect that future versions of `cimg/rust` will change to 24.04 (which may happen 14-Jun-2024). In that case it may just make sense to wait until next week to see if a lot of this complexity just goes away.